File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.py
class Solution:
    def reverseDegree(self, s: str) -> int:
        map = [0] * 26
        n = 26
        for i in range(len(map)):
            map[i] = n
            n -= 1
        
        ans = 0
        for i, v in enumerate(s):
            ans += map[ord(v) - ord('a')] * (i + 1)
        
        return ans
    
    # string
    # time: O(s)
    # space: O(1)

File: 3498-reverse-degree-of-a-string/3498-reverse-degree-of-a-string.go
func reverseDegree(s string) int {
    dict := make([]int, 26)
    n := 26
    for i := 0; i < len(dict); i++ {
        dict[i] = n
        n--
    }

    ans := 0
    for i, r := range s {
        ans += dict[int(r - 'a')] * (i + 1)
    }

    return ans
}

// string
// time: O(s)
// space: O(1)